geoapps/inversion/electricals/driver.py

```diff
@@ -0,0 +1,150 @@
+#  Copyright (c) 2024 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+from __future__ import annotations
+
+import sys
+import uuid
+from copy import deepcopy
+from pathlib import Path
+
+import numpy as np
+from geoh5py.data import Data
+from geoh5py.objects import DrapeModel
+from geoh5py.workspace import Workspace
+
+from geoapps.inversion.components.data import InversionData
+from geoapps.inversion.components.topography import InversionTopography
+from geoapps.inversion.components.windows import InversionWindow
+from geoapps.inversion.line_sweep.driver import LineSweepDriver
+from geoapps.inversion.params import BaseParams
+from geoapps.shared_utils.utils import get_locations, weighted_average
+from geoapps.utils.models import get_drape_model
+from geoapps.utils.surveys import extract_dcip_survey
+
+
+class BasePseudo3DDriver(LineSweepDriver):
+    _params_class: type(BaseParams)
```

Comment:
mypy won't accept BaseParams as without type()?

Reply:
`_param_class` store the type of Params, not the instantiated class.

Reply:
oh, right type(BaseParams) is type class or something.. gotcha!